### DISCUSSION TITLE
Is there a way to introduce a delay between the loops of a DO_WHILE loop or a sleep in netflix conductor?

### DISCUSSION THREAD

**Discussion by jiggy71189** [Netflix/conductor#3054] [Q&A]
I am using Do_While loop in my workflow. And inside loop I have one http task that will be executed again based on the loop Condition. I need to add delay between loop task means whenever it would go for execute second iteration task then it should wait for specific amount of time. 
How can I achieve this feature in workflow?

**Comment by dougsillars** [comment]
Adding a WAIT task can help you solve this:


``` json
{
  "taskType": "WAIT",
  "inputParameters": {
    "duration": "2 days 3 hours"  
  }
}
```
https://orkes.io/content/docs/reference-docs/wait-task

The lowest interval accepted is 1s.

**Reply by dougsillars** [reply]
NOTE: that WAIT tasks have a polling interval.  This is by default 30s, so your step functions on the delay would be 30s.
This loop had a 5s WAIT task in it:
![Timeline of a loop with a wait task inside](https://user-images.githubusercontent.com/1514288/174824039-0250cb54-4364-4d86-99a2-604d49c92145.png)

**Reply by jiggy71189** [reply]
@dougsillars Thanks for the reply. 
Is it possible that we can automatically resume the task of second iteration without rest api call ?

**Reply by dougsillars** [reply]
You can restart the WAIT task with an API call - but in the example above I do not, I simply set a timer, and when that expires, the workflow continues.

To create the diagram in the answer above, I kick off the workflow below with  input  {"sleepytime":5}
![image](https://user-images.githubusercontent.com/1514288/174861758-e5a37b69-9ec9-4cdd-a989-b6205da1a779.png)


Here is the wait task inside the loop.
```json
 {
          "name": "sleeper",
          "taskReferenceName": "sleeper_ref",
          "inputParameters": {
            "duration": "${workflow.input.sleepytime} seconds"
          },
          "type": "WAIT",
        }
```

**Reply by dougsillars** [reply]
https://www.youtube.com/watch?v=6AnMRKZJhio

I made a video to help describe the wait task

**Reply by Vin-terfresh** [reply]
Thank you for the explanation, @dougsillars. Can you tell me, please, how can I change polling interval for wait tasks?
